app/src/main/java/com/io/movio/domain/GetMoviesUseCase.kt

```diff
@@ -1,12 +1,17 @@
 package com.io.movio.domain
 
+import com.io.movio.data.Result
 import com.io.movio.data.models.Movie
 import com.io.movio.data.repositories.MovieRepository
+import java.lang.Exception
 
-class GetMoviesUseCase(private val movieRepository: MovieRepository) {
-    operator fun invoke(): List<Movie> {
-        return getMovies()
-    }
-   fun getMovies(): List<Movie> = movieRepository.getMovies()
+class GetMoviesUseCase : UseCase<MovieRepository, Result<List<Movie>>> {
```

Comment:
The repository is better to be a dependency of the use case. It is a means to an end, but it is not a required data for getting the movies.